### PR TITLE
Use React's `AriaAttributes` type directly rather than recreating

### PR DESCRIPTION
**Pull request by prichey**
Fixes https://github.com/JedWatson/react-select/issues/4940

### CODE REVIEW

File: packages/react-select/src/accessibility/index.ts [hunk 1]
```diff
@@ -11,6 +11,8 @@ export type OptionContext = 'menu' | 'value';
 
 export type GuidanceContext = 'menu' | 'input' | 'value';
 
+export type AriaInvalid = boolean | 'false' | 'true' | 'grammar' | 'spelling';
```
Comment (prichey) [inline packages/react-select/src/accessibility/index.ts:14]: Pulled from here: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L1654

File: packages/react-select/src/Select.tsx [hunk 2]
```diff
@@ -76,7 +77,7 @@ export interface Props<
   /** HTML ID of an element containing an error message related to the input**/
   'aria-errormessage'?: string;
   /** Indicate if the value entered in the field is invalid **/
-  'aria-invalid'?: boolean;
+  'aria-invalid'?: AriaInvalid;
```
Comment (Methuselah96) [inline packages/react-select/src/Select.tsx:80]: Instead of redefining `AriaInvalid`, can we do `AriaAttributes['aria-invalid']` here?

**Review by Methuselah96 — CHANGES_REQUESTED** [review]

File: packages/react-select/src/Select.tsx (hunk 2 above)
Comment (prichey) [reply packages/react-select/src/Select.tsx:80]: Yeah, definitely. I'll make that change now

Comment (prichey) [reply packages/react-select/src/Select.tsx:80]: Do you think it's worth updating the other aria attributes to pull from that type in that case?

Comment (Methuselah96) [reply packages/react-select/src/Select.tsx:80]: If you're up for that, that would be great! Thanks.

File: packages/react-select/src/Select.tsx [hunk 3]
```diff
@@ -74,15 +71,15 @@ export interface Props<
   Group extends GroupBase<Option>
 > {
   /** HTML ID of an element containing an error message related to the input**/
-  'aria-errormessage'?: string;
+  'aria-errormessage'?: AriaAttributes['aria-errormessage'];
```
Comment (prichey) [inline packages/react-select/src/Select.tsx:74]: `AriaAttributes['aria-errormessage']` includes `| undefined` so this is technically equivalent to

```
'aria-errormessage': AriaAttributes['aria-errormessage'];
```

but I'll defer to your preferences here @Methuselah96

File: packages/react-select/src/Select.tsx (hunk 2 above)
Comment (prichey) [reply packages/react-select/src/Select.tsx]: Done ✅

File: packages/react-select/src/Select.tsx (hunk 3 above)
Comment (Methuselah96) [reply packages/react-select/src/Select.tsx:74]: I'm not sure it's always equivalent ([TS playground](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAJQKYEMDG8BmUIjgcilQ3wG4Aoc4AOxiSk3STgHkwZgJqUAbABRxgAznADe5OJLh0hMAPwAuOLKg0A5nAA+cAK7UAJkkw0k+igF9KaHiiEi2HLrzhIAHnQMjk6GADoAwriQ1Ei0ADwOnNz8gkIAfGISUkQG9AAUAJSJUjlwRDA6UNRw1Do8PBQ5lpbkaFyyJVyRTjwAqkIoaswAvHAR7FHOAPRxFFS09IxozAByTQMtAhDC2VIyMEoq6lq6qcYhZuQ11rYic9TN0S7uofpexH6B4FyhMGHnl7xLwgniOSmGKCZVa5PJIApFEplCpJSTVKz1eAwAAW6gATHBeu95o4riMKEA)). Either way, I like the semantic nuance of declaring it as optional.

**Review by Methuselah96 — APPROVED** [review]
Thanks!

**Comment by prichey** [comment]
@Methuselah96 took the liberty to update the only other place we were recreating types from `AriaAttributes`: https://github.com/JedWatson/react-select/pull/4941/commits/9ace36c0b5a83625e589357beec70c71ebdac99f. Going to stop poking at this now! Looking forward to 5.3 😄

**Comment by Methuselah96** [comment]
Thanks! Forgot to ask earlier: can you add changeset?

**Comment by prichey** [comment]
@Methuselah96 Just added one - let me know if I did it correctly, this was my first time using that package

**Comment by Methuselah96** [comment]
Yeah, looks good.

**Comment by prichey** [comment]
@Methuselah96 How do you determine when to cut a new release? It'd be nice to upgrade without having to do my own manual patching. Thanks!

**Comment by Methuselah96** [comment]
@prichey Thanks for the reminder, as you can see we haven't been doing a whole lot of merging/releasing recently. I'll try to get a release out soon. I noticed you made this a minor instead of a patch. Any particular reason?

**Comment by prichey** [comment]
@Methuselah96 No problem! Tbh it could totally be a patch, I think I did that after you had tagged it with 5.3 so I assumed you wanted it to be a minor release. Patch would be awesome if that helps it get released!

**Comment by Methuselah96** [comment]
@prichey Published in [v5.2.2](https://github.com/JedWatson/react-select/releases/tag/react-select%405.2.2).

**Pull request merged**